src/codegate/api/v1.py

```diff
@@ -83,6 +93,30 @@ async def create_workspace(request: v1_models.CreateWorkspaceRequest) -> v1_mode
     return v1_models.Workspace(name=request.name, is_active=False)
 
 
+async def rename_workspace(
+    request: v1_models.CreateOrRenameWorkspaceRequest,
+) -> v1_models.Workspace:
+    try:
+        _ = await wscrud.rename_workspace(request.name, request.rename_to)
+    except crud.WorkspaceDoesNotExistError:
```

Comment:
do we need to add a case for avoiding rename the default one?

Reply:
This is handled in src/codegate/workspaces/crud.py